src/main/java/com/iexec/commons/containers/client/DockerClientInstance.java

```diff
@@ -801,6 +802,56 @@ public synchronized boolean removeContainer(String containerName) {
             return false;
         }
     }
+
+    /**
+     * Retrieves the execution duration of a container.
+     * If the container has not been started yet or has not ended yet,
+     * then return {@link Optional#empty()}.
+     * Otherwise, return the duration.
+     * <p>
+     * /!\ Docker inspection command precision could lead to sub-zero execution duration for fast containers.
+     * In this case, this method return a zero-duration object.
+     *
+     * @param containerName Name of the container to look for execution duration.
+     * @return {@code Optional#empty()} if not started or ended,
+     * the duration otherwise.
+     */
+    public Optional<Duration> getContainerExecutionDuration(String containerName) {
+        try (InspectContainerCmd inspectContainerCmd = getClient().inspectContainerCmd(containerName)) {
+            final InspectContainerResponse.ContainerState state = inspectContainerCmd.exec().getState();
+            return getContainerExecutionDuration(containerName, state.getStartedAt(), state.getFinishedAt());
+        } catch (DockerException e) {
+            log.warn("Can't get execution duration of container [containerName:{}]", containerName, e);
+            return Optional.empty();
+        }
+    }
+
+    Optional<Duration> getContainerExecutionDuration(String containerName,
+                                                            String startedAt,
+                                                            String finishedAt) {
+        final String defaultTime = "0001-01-01T00:00:00Z";
+        if (defaultTime.equals(startedAt)) {
+            log.debug("Container has not been started yet [containerName:{}]", containerName);
+            return Optional.empty();
+        }
+        if (defaultTime.equals(finishedAt)) {
+            log.debug("Container has not been ended yet [containerName:{}]", containerName);
+            return Optional.empty();
+        }
+
+        final Instant startDate = Instant.parse(startedAt);
+        final Instant endDate = Instant.parse(finishedAt);
+
+        final Duration duration = Duration.between(startDate, endDate);
+        if (duration.isNegative()) {
+            // This could mean the command was wrong and has not been correctly executed
+            log.debug("Container has finished faster than Docker precision [containerName:{}]", containerName);
```

Comment:
Do we want this to only be debug or could it be a warning ?
```suggestion
            log.warn("Container has finished faster than Docker precision [containerName:{}]", containerName);
```

Reply:
good to me, we can drop this suggestion